integration/src/test/java/dev/ikm/tinkar/integration/search/SearcherIT.java

```diff
@@ -160,7 +167,87 @@ public void searchFromDescendantsOfConceptWithCustomCalculatorIT() throws Except
     }
 
     @Test
-    public void searchConceptsNonExistingMembershipSemantic() throws Exception {
+    public void typeAheadIndexerTest() throws Exception {
+        List<String> suggestions = TypeAheadSearch.suggest("r");
+        assertEquals(40, suggestions.size());
+
+        // Add a new semantic
+        MutableList<String> list = Lists.mutable.empty();
+        list.add("rAdded");
+        Session session = composer.open(State.ACTIVE, TinkarTerm.USER, TinkarTerm.SOLOR_OVERLAY_MODULE, TinkarTerm.DEVELOPMENT_PATH);
+        session.compose((SemanticAssembler semanticAssembler) -> semanticAssembler
+                .pattern(TinkarTerm.COMMENT_PATTERN)
+                .reference(TinkarTerm.COMMENT)
+                .fieldValues((MutableList<Object> values) -> values
+                        .withAll(list))
+        );
+        suggestions = TypeAheadSearch.suggest("r");
+        assertEquals(41, suggestions.size());
+    }
+
+    @Test
+    public void typeAheadIndexerTestSearchAndDescendants1() throws Exception {
+        // f7495b58-6630-3499-a44e-2052b5fcf06c - Author ID
+        // Model concept: [7bbd4210-381c-11e7-9598-0800200c9a66]
+
+        Stopwatch stopwatch = new Stopwatch();
+        PublicId ancestorId = EntityService.get().getEntity(UUID.fromString("f7495b58-6630-3499-a44e-2052b5fcf06c")).get().publicId();
+
+        String userInput = "u";
+        List<LatestVersionSearchResult> allSearchResults = TypeAheadSearch.typeAheadSuggestions(userInput, ancestorId);
+
+        stopwatch.stop();
+    }
+
+    @Test
+    public void typeAheadIndexerTestSearchAndDescendants2() throws Exception {
+        // f7495b58-6630-3499-a44e-2052b5fcf06c - Author ID
+        // Model concept: [7bbd4210-381c-11e7-9598-0800200c9a66]
+
+        long startTime = System.nanoTime();
+
+        PublicId ancestorId = EntityService.get().getEntity(UUID.fromString("7bbd4210-381c-11e7-9598-0800200c9a66")).get().publicId();
+
+
+        String userInput = "a";
+        List<LatestVersionSearchResult> allSearchResults = TypeAheadSearch.typeAheadSuggestions(userInput, ancestorId);
+
+        long endTime = System.nanoTime();
+        long duration = endTime - startTime;
+    }
+
+    @Test
+    public void typeAheadIndexerTestSearchAndDescendantsFuzzy() throws Exception {
+        // f7495b58-6630-3499-a44e-2052b5fcf06c - Author ID
+        // Model concept: [7bbd4210-381c-11e7-9598-0800200c9a66]
+        PublicId ancestorId = EntityService.get().getEntity(UUID.fromString("f7495b58-6630-3499-a44e-2052b5fcf06c")).get().publicId();
+
+        Stopwatch stopwatch = new Stopwatch();
+
+        String userInput = "u";
+        List<LatestVersionSearchResult> allSearchResults = TypeAheadSearch.typeAheadFuzzySuggestions(userInput, ancestorId);
+
+        stopwatch.stop();
+    }
+
+    @Test
+    public void typeAheadIndexerTestSearchAndDescendantsFuzzy2() throws Exception {
```

Comment:
I get this exception when running this test FYI:

`java.lang.IllegalStateException: No DataServiceController selected for provider. Tried 0`